### ISSUE TITLE
AppBundleHelper hangs after Google Play Store update.

### ISSUE THREAD

**Issue by nickdonnellyriis** [dji-sdk/Mobile-SDK-Android#642]
If you install the mobile SDK as a feature module, then push it to Google Play, it works fine on the initial install and we can register the SDK.

However, if we push an update to the Play store (just a version number bump, with no other changes), the app hangs indefinitely when calling `AppBundleHelper.init()`. Our code looks like this:

```
        SplitCompat.installActivity(requireActivity())
        SplitCompat.install(context.applicationContext)
        Helper.install(context.applicationContext as Application)
        AppBundleHelper.init()
```

Is this a known issue? And are there any workarounds? This is a major issue as we are unable to push any updates to our application.

DJI SDK Version: tested on 4.13 and 4.13.1

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/40451'>Zendesk ticket #40451</a>:<br><br>Dear Client
Thank you for contacting DJI.

We havn't see this issue before, Can you try push the sample app to google store and try it? If the issue replicate then its our side issue, otherwise its your side.

Hopefully our solution can help you.
Kindly Regards,
DJI Developer Support

**Comment by nickdonnellyriis** [comment]
I will give it as hot with the sample app.

We have narrowed it down exactly to the `AppBundleHelper.init()` call. What does that call actually do internally? It might be helpful in pinpointing the issue...

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/40451'>Zendesk ticket #40451</a>:<br><br>Dear Client
Thank you for contacting DJI.

is AppBundleHelper.init exists in our sample code? I cannot find it. It seems like android api but I cannot find it on Google which is weird. I have look at the sample code and in the app register segment, the code is like below.
 ![](https://djisdksupport.zendesk.com/attachments/token/kjCmNN4sH6taY4yHRcCiUs4te/?name=inline1652798839.png)​

Hopefully our solution can help you.
Kindly Regards,
DJI Developer Support<br><a href='https://djisdksupport.zendesk.com/attachments/token/kjCmNN4sH6taY4yHRcCiUs4te' target='zd_attach'>inline1652798839.png</a>

**Comment by nickdonnellyriis** [comment]
Yes, it exists in your code in the dynamic module here: https://github.com/DJI-Mobile-SDK-Tutorials/DJI-Android-SDK-Bundle-Sample/blob/a3e0a1a0fc4762c7db8363ee6060c541d54b3aef/sample/dynamic_dji_sdk/src/main/java/com/dji/sdk/sample/internal/controller/MainActivity.java#L103

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/40451'>Zendesk ticket #40451</a>:<br><br>Dear Client
Thank you for contacting DJI.

Thank you very much for your information, so the problem is before your uploading the app to Google, it workl  fine. After you have uploading the apk to Google and you download/install the app, you fine the app hang on AppBundleHelper.init(). Am I correct? I will ask the dev team and respond to you once I have got the feedback.

Hopefully our solution can help you.
Kindly Regards,
DJI Developer Support

**Comment by dji-dev** [comment]
Agent comment from William Wong in <a href='https://djisdksupport.zendesk.com/agent/tickets/40451'>Zendesk ticket #40451</a>:<br><br>Dear Client
Thank you for contacting DJI.

I feel like its the debug and release apk difference. Can you try on debug mode and release mode to see if there is any difference?

Hopefully our solution can help you.
Kindly Regards,
DJI Developer Support

**Comment by nickdonnellyriis** [comment]
We will give a debug aab a try. But essentially, yes. The steps we are going through are:

1. Build a signed app bundle (aab)
2. Upload the aab to play store
3. Download it on device and connect a DJI drone (works fine)
4. Increase the version code of the app
5. Build a new aab (nothing else changed but the version)
6. Upload the new aab to play store as an update
7. Update the app on the phone through play store
8. Loading the DJI sdk dynamically hangs on the app bundle helper call.

**Comment by nickdonnellyriis** [comment]
@dji-dev Google Play does not allow us to upload a `debug` version of the app.

**Comment by kenargo** [comment]
Set ‘minifyEnabled false’ in the release buildTypes. 

 

You actually can upload debug to the store but I’m guessing your signing cert for debug is the wrong cert.

**Comment by nickdonnellyriis** [comment]
@kenargo `minifyEnabled` is already set to false for our release variants.

**Comment by nickdonnellyriis** [comment]
We've gone ahead and replaced our dynamic module with the one from your sample, and we are still observing freezing after the app has updated. Is there anything package specific to `AppBundleHelper.init()`?

**Comment by kenargo** [comment]
My bad, minify is one thing and R8 (Proguard) is another.

 

Try:

 

 

useProguard false

**Comment by kenargo** [comment]
Hit enter/send before I finished:

 

You might also need to add these rules to your proguard file:

 

-dontshrink

-dontobfuscate

-dontoptimize

 

 

From: kenargo@kenargo.com <kenargo@kenargo.com> 
Sent: Thursday, October 29, 2020 4:44 PM
To: 'dji-sdk/Mobile-SDK-Android' <reply@reply.github.com>; 'dji-sdk/Mobile-SDK-Android' <Mobile-SDK-Android@noreply.github.com>
Cc: 'Mention' <mention@noreply.github.com>
Subject: RE: [dji-sdk/Mobile-SDK-Android] AppBundleHelper hangs after Google Play Store update. (#642)

 

My bad, minify is one thing and R8 (Proguard) is another.

 

Try:

 

 

useProguard false

**Comment by luoyexk** [comment]
If you want to test your *.aab can run on the device successfully, there are two ways:

1. [Official Document](https://developer.android.com/guide/app-bundle/test)
Deploy aab when click `Run app`, 
![image](https://user-images.githubusercontent.com/20855787/97656875-792e4900-1aa3-11eb-8bbd-73aad0e9a1c4.png)

2. Build an aab file, then use the bundletool to install it. [bundletool]( https://developer.android.com/studio/command-line/bundletool)

Unfortunately, if you use method 1 to try to build the DJI sample code(v4.13.1) then install on the app, the app will ANR on application start!! Change deploy from `APK from app bundle` to `Default APK`, it will work well.

**Comment by nickdonnellyriis** [comment]
@kenargo thanks for the suggestion. The problem seems to remain even with the proguard configuration changes.

@luoyexk The app bundle works perfectly fine locally. It is only when we push to the google play store *and then update the app* that it ANRs.

**Comment by nickdonnellyriis** [comment]
We've narrowed this down to something very bizarre:

We have a separate library that we've included in the project that depends on `com.mapbox.mapboxsdk:mapbox-android-sdk:9.2.1` - removing this dependency fixes the issue. Downgrading to earlier versions of mapbox, the crash is still present.

This dependency is absolutely required on our end, why would the app bundle init call be interfering with this?

**Comment by nickdonnellyriis** [comment]
This seems to have something to do with #222 

Unfortunately, the suggested fix of removing `arm64-v8a` ABI is not an option, as the Google Play Store now requires all apps to support 64-bit.

**Comment by nickdonnellyriis** [comment]
Any new information on this? It's a huge problem.

**Comment by furick1** [comment]
@nickdonnellyriis we have the exact same issue. this is killing us right now. do we need to ask mapbox too?

**Comment by DavisDevasia** [comment]
@dji-dev We have a similar issue too. APK installations are running fine, but the app goes to ANR when installed from Google Play store or when installed from local AAB file.

**Comment by luoyexk** [comment]
@DavisDevasia 
If your gradle plugin version is above 3.3.0
, add a line of code `android.bundle.enableUncompressedNativeLibs=false` in  `gradle.properties`.

And the gradle plugin version is above 3.6.0, add a line of code in `AndroidMenifest.xml` ->
```
<application
        android:extractNativeLibs="true"
        ...
>
       ...
</>

```

See the new feature in Gradle v3.3.0 [https://developer.android.com/studio/releases/gradle-plugin#behavior-changes](https://developer.android.com/studio/releases/gradle-plugin#behavior-changes)

new feature in Gradle v3.6.0 [https://developer.android.com/studio/releases/gradle-plugin#3-6-0](https://developer.android.com/studio/releases/gradle-plugin#3-6-0)

**Comment by luoyexk** [comment]
project build.gradle

`classpath 'com.android.tools.build:gradle:4.0.1'`

gradle wrapper

`distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip`

Other versions are not guaranteed to be ANR.

**Comment by furick1** [comment]
android.bundle.enableUncompressedNativeLibs=false worked!  Thank you!!!  yo @nickdonnellyriis give it a shot on your end.

**Comment by DavisDevasia** [comment]
Added `android.bundle.enableUncompressedNativeLibs=false` in `gradle.properties` and it works fine.
I think I'll add `android:extractNativeLibs="true"' in `AndroidMenifest.xml` to be safe.

**Comment by MeteC** [comment]
@luoyexk Thank the heavens for you sir! I've wasted days and days trying to track this same issue down intermittently since 2019. And several moments of sheer panic as a newly released app bundle just freezes on my users! 

@dji-dev - for god's sake your SDK doesn't work easily with the (now required) app bundle packaging, PLEASE tell developers how to use it correctly on your developer website. You've got so many tutorials there, surely you can add the information in! We shouldn't be hunting through old github issues to find solutions!
